internal/providers/pluginfw/resources/user/data_users.go

```diff
@@ -0,0 +1,113 @@
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+)
+
+func DataSourceUsers() datasource.DataSource {
+	return &UsersDataSource{}
+}
+
+var _ datasource.DataSourceWithConfigure = &UsersDataSource{}
+
+type UsersDataSource struct {
+	Client *common.DatabricksClient
+}
+
+type UsersInfo struct {
+	DisplayNameContains string     `json:"display_name_contains,omitempty" tf:"computed"`
+	UserNameContains    string     `json:"user_name_contains,omitempty" tf:"computed"`
+	Users               []iam.User `json:"users,omitempty" tf:"computed"`
+}
+
+func (d *UsersDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = "databricks_users"
+}
+
+func (d *UsersDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	attrs, blocks := tfschema.DataSourceStructToSchemaMap(UsersInfo{}, nil)
+	resp.Schema = schema.Schema{
+		Attributes: attrs,
+		Blocks:     blocks,
+	}
+}
+
+func (d *UsersDataSource) Configure(_ context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	if d.Client == nil {
+		d.Client = pluginfwcommon.ConfigureDataSource(req, resp)
+	}
+}
+
+// AppendDiagAndCheckErrors is a helper function that simplifies error handling by combining the appending of diagnostics and the checking of errors in a single step.
+// It is particularly useful in data source and resource read operations where you want to append diagnostics and immediately determine if an error has occurred.
+func AppendDiagAndCheckErrors(resp *datasource.ReadResponse, diags diag.Diagnostics) bool {
+	resp.Diagnostics.Append(diags...)
+	return resp.Diagnostics.HasError()
+}
+
+// DiagsFromError helps us create an error diagnostic from an error.
+func DiagsFromError(summary string, err error) diag.Diagnostics {
+	return diag.Diagnostics{
+		diag.NewErrorDiagnostic(summary, err.Error()),
+	}
+}
+
+func validateFilters(data *UsersInfo) diag.Diagnostics {
+	if data.DisplayNameContains != "" && data.UserNameContains != "" {
+		return diag.Diagnostics{diag.NewErrorDiagnostic("Invalid configuration", "Exactly one of display_name_contains or user_name_contains should be specified, not both.")}
+	}
+	return nil
+}
+
+func (d *UsersDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var userInfo UsersInfo
+
+	diags := req.Config.Get(ctx, &userInfo)
+	diags = append(diags, validateFilters(&userInfo)...)
+	if AppendDiagAndCheckErrors(resp, diags) {
+		return
+	}
+
+	var filter string
+
+	if userInfo.DisplayNameContains != "" {
+		filter = fmt.Sprintf("displayName co \"%s\"", userInfo.DisplayNameContains)
+	} else if userInfo.UserNameContains != "" {
+		filter = fmt.Sprintf("userName co \"%s\"", userInfo.UserNameContains)
+	}
+
+	var users []iam.User
+	var err error
+
+	if d.Client.Config.IsAccountClient() {
+		a, diags := d.Client.GetAccountClient()
+		if AppendDiagAndCheckErrors(resp, diags) {
```

Comment:
Let's follow the code convention we've established here for error handling: https://github.com/databricks/terraform-provider-databricks/blob/main/CONTRIBUTING.md#code-conventions. I think we don't need AppendDiagAndCheckErrors or DiagsFromError, as we cover how to handle methods that return both `diag.Diagnostics` and `error` there.

Reply:
Makes sense, done! 👍